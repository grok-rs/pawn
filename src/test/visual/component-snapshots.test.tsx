import React from 'react';
import { render } from '@testing-library/react';
import { createMockTournament, createMockPlayer } from '../utils/test-utils';

// Mock components for visual testing - import actual components as needed
// These would normally import the real components from your app
const MockButton = ({
  children,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  ...props
}: any) => (
  <button
    className={`btn btn-${variant} btn-${size} ${disabled ? 'btn-disabled' : ''}`}
    disabled={disabled}
    {...props}
  >
    {children}
  </button>
);

const MockStatCard = ({
  title,
  value,
  subtitle,
  trend,
  color = 'primary',
  ...props
}: any) => (
  <div className={`stat-card stat-card-${color}`} {...props}>
    <div className="stat-card-header">
      <h3 className="stat-card-title">{title}</h3>
      {trend && (
        <span
          className={`stat-card-trend ${trend > 0 ? 'positive' : 'negative'}`}
        >
          {trend}%
        </span>
      )}
    </div>
    <div className="stat-card-value">{value}</div>
    {subtitle && <div className="stat-card-subtitle">{subtitle}</div>}
  </div>
);

const MockEmptyState = ({
  title,
  description,
  action,
  icon,
  ...props
}: any) => (
  <div className="empty-state" {...props}>
    {icon && <div className="empty-state-icon">{icon}</div>}
    <h3 className="empty-state-title">{title}</h3>
    <p className="empty-state-description">{description}</p>
    {action && <div className="empty-state-action">{action}</div>}
  </div>
);

const MockLoadingButton = ({ loading = false, children, ...props }: any) => (
  <button
    className={`loading-btn ${loading ? 'loading' : ''}`}
    disabled={loading}
    {...props}
  >
    {loading && <span className="spinner" />}
    <span>{children}</span>
  </button>
);

// Visual testing utilities

const createResponsiveVisualTest = (
  name: string,
  component: React.ReactElement
) => {
  const breakpoints = [
    { name: 'mobile', width: 375 },
    { name: 'tablet', width: 768 },
    { name: 'desktop', width: 1024 },
    { name: 'large-desktop', width: 1440 },
  ];

  breakpoints.forEach(({ name: breakpointName, width }) => {
    test(`${name} - ${breakpointName} (${width}px) visual snapshot`, () => {
      // Mock window dimensions
      Object.defineProperty(window, 'innerWidth', {
        value: width,
        configurable: true,
      });
      Object.defineProperty(window, 'innerHeight', {
        value: 800,
        configurable: true,
      });

      // Trigger resize event
      window.dispatchEvent(new Event('resize'));

      const { container } = render(component);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
};

const createStateVisualTest = (
  name: string,
  componentStates: Array<{ stateName: string; component: React.ReactElement }>
) => {
  componentStates.forEach(({ stateName, component }) => {
    test(`${name} - ${stateName} state visual snapshot`, () => {
      const { container } = render(component);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
};

describe('Visual Regression Tests - Component Snapshots', () => {
  describe('Button Component', () => {
    createStateVisualTest('Button', [
      {
        stateName: 'primary',
        component: <MockButton variant="primary">Primary Button</MockButton>,
      },
      {
        stateName: 'secondary',
        component: (
          <MockButton variant="secondary">Secondary Button</MockButton>
        ),
      },
      {
        stateName: 'success',
        component: <MockButton variant="success">Success Button</MockButton>,
      },
      {
        stateName: 'danger',
        component: <MockButton variant="danger">Danger Button</MockButton>,
      },
      {
        stateName: 'disabled',
        component: <MockButton disabled>Disabled Button</MockButton>,
      },
      {
        stateName: 'small',
        component: <MockButton size="small">Small Button</MockButton>,
      },
      {
        stateName: 'large',
        component: <MockButton size="large">Large Button</MockButton>,
      },
    ]);

    createResponsiveVisualTest(
      'Button Responsive',
      <MockButton variant="primary">Responsive Button</MockButton>
    );
  });

  describe('StatCard Component', () => {
    createStateVisualTest('StatCard', [
      {
        stateName: 'basic',
        component: (
          <MockStatCard
            title="Total Players"
            value="156"
            subtitle="Registered this month"
          />
        ),
      },
      {
        stateName: 'with-trend-positive',
        component: (
          <MockStatCard
            title="Active Tournaments"
            value="12"
            subtitle="Currently running"
            trend={15}
            color="success"
          />
        ),
      },
      {
        stateName: 'with-trend-negative',
        component: (
          <MockStatCard
            title="Completion Rate"
            value="87%"
            subtitle="This week"
            trend={-3}
            color="warning"
          />
        ),
      },
      {
        stateName: 'large-numbers',
        component: (
          <MockStatCard
            title="Total Games Played"
            value="23,456"
            subtitle="Since launch"
            color="primary"
          />
        ),
      },
    ]);
  });

  describe('EmptyState Component', () => {
    createStateVisualTest('EmptyState', [
      {
        stateName: 'basic',
        component: (
          <MockEmptyState
            title="No tournaments yet"
            description="Create your first tournament to get started."
          />
        ),
      },
      {
        stateName: 'with-action',
        component: (
          <MockEmptyState
            title="No players found"
            description="Add players to this tournament to begin pairing rounds."
            action={<MockButton variant="primary">Add Players</MockButton>}
          />
        ),
      },
      {
        stateName: 'with-icon',
        component: (
          <MockEmptyState
            title="No results available"
            description="Complete some games to see tournament standings."
            icon="🏆"
          />
        ),
      },
    ]);
  });

  describe('LoadingButton Component', () => {
    createStateVisualTest('LoadingButton', [
      {
        stateName: 'idle',
        component: <MockLoadingButton>Save Tournament</MockLoadingButton>,
      },
      {
        stateName: 'loading',
        component: <MockLoadingButton loading>Saving...</MockLoadingButton>,
      },
    ]);
  });

  describe('Data Display Components', () => {
    test('Player profile card - visual snapshot', () => {
      const mockPlayer = createMockPlayer({
        name: 'Alice Johnson',
        rating: 1650,
        title: 'WFM',
        countryCode: 'US',
      });

      const PlayerCard = ({ player }: { player: any }) => (
        <div className="player-card">
          <div className="player-card-header">
            <div className="player-name">
              {player.title && (
                <span className="player-title">{player.title}</span>
              )}
              {player.name}
            </div>
            <div className="player-country">{player.countryCode}</div>
          </div>
          <div className="player-rating">{player.rating}</div>
          <div className="player-details">
            <span>Email: {player.email}</span>
            <span>Phone: {player.phone}</span>
          </div>
        </div>
      );

      const { container } = render(<PlayerCard player={mockPlayer} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Tournament card - visual snapshot', () => {
      const mockTournament = createMockTournament({
        name: 'Spring Championship 2024',
        playerCount: 32,
        status: 'active',
        maxRounds: 7,
      });

      const TournamentCard = ({ tournament }: { tournament: any }) => (
        <div className="tournament-card">
          <div className="tournament-card-header">
            <h3 className="tournament-name">{tournament.name}</h3>
            <span className={`tournament-status status-${tournament.status}`}>
              {tournament.status}
            </span>
          </div>
          <div className="tournament-stats">
            <div className="stat">
              <span className="stat-label">Players</span>
              <span className="stat-value">{tournament.playerCount}</span>
            </div>
            <div className="stat">
              <span className="stat-label">Rounds</span>
              <span className="stat-value">
                {tournament.rounds}/{tournament.maxRounds}
              </span>
            </div>
          </div>
          <div className="tournament-actions">
            <MockButton variant="primary" size="small">
              View Details
            </MockButton>
            <MockButton variant="secondary" size="small">
              Export
            </MockButton>
          </div>
        </div>
      );

      const { container } = render(
        <TournamentCard tournament={mockTournament} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Form Components', () => {
    test('Tournament form - visual snapshot', () => {
      const TournamentForm = () => (
        <form className="tournament-form">
          <div className="form-section">
            <h3>Basic Information</h3>
            <div className="form-field">
              <label>Tournament Name</label>
              <input
                type="text"
                className="form-input"
                defaultValue="Spring Championship 2024"
              />
            </div>
            <div className="form-field">
              <label>Description</label>
              <textarea
                className="form-textarea"
                rows={3}
                defaultValue="Annual spring tournament"
              />
            </div>
          </div>

          <div className="form-section">
            <h3>Tournament Settings</h3>
            <div className="form-row">
              <div className="form-field">
                <label>Max Players</label>
                <input type="number" className="form-input" defaultValue="16" />
              </div>
              <div className="form-field">
                <label>Max Rounds</label>
                <input type="number" className="form-input" defaultValue="5" />
              </div>
            </div>
            <div className="form-field">
              <label>Pairing Method</label>
              <select className="form-select" defaultValue="swiss">
                <option value="swiss">Swiss System</option>
                <option value="round_robin">Round Robin</option>
                <option value="knockout">Knockout</option>
              </select>
            </div>
          </div>

          <div className="form-actions">
            <MockButton variant="secondary">Cancel</MockButton>
            <MockButton variant="primary">Create Tournament</MockButton>
          </div>
        </form>
      );

      const { container } = render(<TournamentForm />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Player form - visual snapshot', () => {
      const PlayerForm = () => (
        <form className="player-form">
          <div className="form-section">
            <h3>Personal Information</h3>
            <div className="form-row">
              <div className="form-field">
                <label>Full Name</label>
                <input
                  type="text"
                  className="form-input"
                  defaultValue="Alice Johnson"
                />
              </div>
              <div className="form-field">
                <label>Title</label>
                <select className="form-select" defaultValue="WFM">
                  <option value="">No Title</option>
                  <option value="CM">CM</option>
                  <option value="FM">FM</option>
                  <option value="WFM">WFM</option>
                  <option value="IM">IM</option>
                  <option value="GM">GM</option>
                </select>
              </div>
            </div>

            <div className="form-row">
              <div className="form-field">
                <label>Rating</label>
                <input
                  type="number"
                  className="form-input"
                  defaultValue="1650"
                />
              </div>
              <div className="form-field">
                <label>Country</label>
                <select className="form-select" defaultValue="US">
                  <option value="US">United States</option>
                  <option value="CA">Canada</option>
                  <option value="UK">United Kingdom</option>
                </select>
              </div>
            </div>
          </div>

          <div className="form-section">
            <h3>Contact Information</h3>
            <div className="form-field">
              <label>Email</label>
              <input
                type="email"
                className="form-input"
                defaultValue="alice@example.com"
              />
            </div>
            <div className="form-field">
              <label>Phone</label>
              <input
                type="tel"
                className="form-input"
                defaultValue="+1 (555) 123-4567"
              />
            </div>
          </div>

          <div className="form-actions">
            <MockButton variant="secondary">Cancel</MockButton>
            <MockButton variant="primary">Save Player</MockButton>
          </div>
        </form>
      );

      const { container } = render(<PlayerForm />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Layout Components', () => {
    test('Navigation header - visual snapshot', () => {
      const Navigation = () => (
        <header className="app-header">
          <div className="nav-brand">
            <h1>♜ Pawn</h1>
          </div>
          <nav className="nav-menu">
            <a href="/tournaments" className="nav-link active">
              Tournaments
            </a>
            <a href="/players" className="nav-link">
              Players
            </a>
            <a href="/settings" className="nav-link">
              Settings
            </a>
          </nav>
          <div className="nav-actions">
            <MockButton variant="primary" size="small">
              New Tournament
            </MockButton>
          </div>
        </header>
      );

      const { container } = render(<Navigation />);
      expect(container.firstChild).toMatchSnapshot();
    });

    createResponsiveVisualTest(
      'Navigation Responsive',
      <header className="app-header">
        <div className="nav-brand">
          <h1>♜ Pawn</h1>
        </div>
        <nav className="nav-menu">
          <a href="/tournaments" className="nav-link active">
            Tournaments
          </a>
          <a href="/players" className="nav-link">
            Players
          </a>
          <a href="/settings" className="nav-link">
            Settings
          </a>
        </nav>
      </header>
    );

    test('Sidebar layout - visual snapshot', () => {
      const SidebarLayout = () => (
        <div className="app-layout">
          <aside className="sidebar">
            <div className="sidebar-header">
              <h2>Tournament Menu</h2>
            </div>
            <nav className="sidebar-nav">
              <a href="#overview" className="sidebar-link active">
                Overview
              </a>
              <a href="#players" className="sidebar-link">
                Players
              </a>
              <a href="#pairings" className="sidebar-link">
                Pairings
              </a>
              <a href="#results" className="sidebar-link">
                Results
              </a>
              <a href="#standings" className="sidebar-link">
                Standings
              </a>
            </nav>
          </aside>
          <main className="main-content">
            <div className="content-header">
              <h1>Tournament Overview</h1>
              <MockButton variant="primary">Edit Tournament</MockButton>
            </div>
            <div className="content-body">
              <p>Main content area</p>
            </div>
          </main>
        </div>
      );

      const { container } = render(<SidebarLayout />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Theme Variations', () => {
    test('Dark theme components - visual snapshot', () => {
      const DarkThemeDemo = () => (
        <div className="theme-dark">
          <div className="card">
            <h3>Dark Theme Card</h3>
            <p>This demonstrates dark theme styling</p>
            <div className="button-group">
              <MockButton variant="primary">Primary</MockButton>
              <MockButton variant="secondary">Secondary</MockButton>
            </div>
          </div>
          <MockStatCard
            title="Dark Theme Stats"
            value="42"
            subtitle="Looking good in dark mode"
            color="primary"
          />
        </div>
      );

      const { container } = render(<DarkThemeDemo />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('High contrast theme - visual snapshot', () => {
      const HighContrastDemo = () => (
        <div className="theme-high-contrast">
          <div className="card">
            <h3>High Contrast Theme</h3>
            <p>Accessibility-focused high contrast theme</p>
            <div className="button-group">
              <MockButton variant="primary">Primary Action</MockButton>
              <MockButton variant="secondary">Secondary Action</MockButton>
            </div>
          </div>
          <MockEmptyState
            title="High Contrast Empty State"
            description="Clear visibility for all users"
            action={<MockButton variant="primary">Take Action</MockButton>}
          />
        </div>
      );

      const { container } = render(<HighContrastDemo />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Error States', () => {
    test('Error message components - visual snapshot', () => {
      const ErrorStates = () => (
        <div className="error-states">
          <div className="error-banner error-banner-danger">
            <strong>Error:</strong> Failed to save tournament data.
          </div>
          <div className="error-banner error-banner-warning">
            <strong>Warning:</strong> Some player ratings are missing.
          </div>
          <div className="error-banner error-banner-info">
            <strong>Info:</strong> Tournament will start in 30 minutes.
          </div>
          <div className="error-inline">
            <span className="error-text">Invalid email format</span>
          </div>
        </div>
      );

      const { container } = render(<ErrorStates />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
