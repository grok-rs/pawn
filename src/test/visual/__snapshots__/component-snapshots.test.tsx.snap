// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - danger state visual snapshot 1`] = `
<button
  class="btn btn-danger btn-medium "
>
  Danger Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - disabled state visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium btn-disabled"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - large state visual snapshot 1`] = `
<button
  class="btn btn-primary btn-large "
>
  Large Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - primary state visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium "
>
  Primary Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - secondary state visual snapshot 1`] = `
<button
  class="btn btn-secondary btn-medium "
>
  Secondary Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - small state visual snapshot 1`] = `
<button
  class="btn btn-primary btn-small "
>
  Small Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button - success state visual snapshot 1`] = `
<button
  class="btn btn-success btn-medium "
>
  Success Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button Responsive - desktop (1024px) visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium "
>
  Responsive Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button Responsive - large-desktop (1440px) visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium "
>
  Responsive Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button Responsive - mobile (375px) visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium "
>
  Responsive Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Button Component > Button Responsive - tablet (768px) visual snapshot 1`] = `
<button
  class="btn btn-primary btn-medium "
>
  Responsive Button
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > Data Display Components > Player profile card - visual snapshot 1`] = `
<div
  class="player-card"
>
  <div
    class="player-card-header"
  >
    <div
      class="player-name"
    >
      <span
        class="player-title"
      >
        WFM
      </span>
      Alice Johnson
    </div>
    <div
      class="player-country"
    >
      US
    </div>
  </div>
  <div
    class="player-rating"
  >
    1650
  </div>
  <div
    class="player-details"
  >
    <span>
      Email: 
      test@example.com
    </span>
    <span>
      Phone: 
      +1234567890
    </span>
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > Data Display Components > Tournament card - visual snapshot 1`] = `
<div
  class="tournament-card"
>
  <div
    class="tournament-card-header"
  >
    <h3
      class="tournament-name"
    >
      Spring Championship 2024
    </h3>
    <span
      class="tournament-status status-active"
    >
      active
    </span>
  </div>
  <div
    class="tournament-stats"
  >
    <div
      class="stat"
    >
      <span
        class="stat-label"
      >
        Players
      </span>
      <span
        class="stat-value"
      >
        32
      </span>
    </div>
    <div
      class="stat"
    >
      <span
        class="stat-label"
      >
        Rounds
      </span>
      <span
        class="stat-value"
      >
        0
        /
        7
      </span>
    </div>
  </div>
  <div
    class="tournament-actions"
  >
    <button
      class="btn btn-primary btn-small "
    >
      View Details
    </button>
    <button
      class="btn btn-secondary btn-small "
    >
      Export
    </button>
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > EmptyState Component > EmptyState - basic state visual snapshot 1`] = `
<div
  class="empty-state"
>
  <h3
    class="empty-state-title"
  >
    No tournaments yet
  </h3>
  <p
    class="empty-state-description"
  >
    Create your first tournament to get started.
  </p>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > EmptyState Component > EmptyState - with-action state visual snapshot 1`] = `
<div
  class="empty-state"
>
  <h3
    class="empty-state-title"
  >
    No players found
  </h3>
  <p
    class="empty-state-description"
  >
    Add players to this tournament to begin pairing rounds.
  </p>
  <div
    class="empty-state-action"
  >
    <button
      class="btn btn-primary btn-medium "
    >
      Add Players
    </button>
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > EmptyState Component > EmptyState - with-icon state visual snapshot 1`] = `
<div
  class="empty-state"
>
  <div
    class="empty-state-icon"
  >
    üèÜ
  </div>
  <h3
    class="empty-state-title"
  >
    No results available
  </h3>
  <p
    class="empty-state-description"
  >
    Complete some games to see tournament standings.
  </p>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > Error States > Error message components - visual snapshot 1`] = `
<div
  class="error-states"
>
  <div
    class="error-banner error-banner-danger"
  >
    <strong>
      Error:
    </strong>
     Failed to save tournament data.
  </div>
  <div
    class="error-banner error-banner-warning"
  >
    <strong>
      Warning:
    </strong>
     Some player ratings are missing.
  </div>
  <div
    class="error-banner error-banner-info"
  >
    <strong>
      Info:
    </strong>
     Tournament will start in 30 minutes.
  </div>
  <div
    class="error-inline"
  >
    <span
      class="error-text"
    >
      Invalid email format
    </span>
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > Form Components > Player form - visual snapshot 1`] = `
<form
  class="player-form"
>
  <div
    class="form-section"
  >
    <h3>
      Personal Information
    </h3>
    <div
      class="form-row"
    >
      <div
        class="form-field"
      >
        <label>
          Full Name
        </label>
        <input
          class="form-input"
          type="text"
          value="Alice Johnson"
        />
      </div>
      <div
        class="form-field"
      >
        <label>
          Title
        </label>
        <select
          class="form-select"
        >
          <option
            value=""
          >
            No Title
          </option>
          <option
            value="CM"
          >
            CM
          </option>
          <option
            value="FM"
          >
            FM
          </option>
          <option
            selected=""
            value="WFM"
          >
            WFM
          </option>
          <option
            value="IM"
          >
            IM
          </option>
          <option
            value="GM"
          >
            GM
          </option>
        </select>
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-field"
      >
        <label>
          Rating
        </label>
        <input
          class="form-input"
          type="number"
          value="1650"
        />
      </div>
      <div
        class="form-field"
      >
        <label>
          Country
        </label>
        <select
          class="form-select"
        >
          <option
            selected=""
            value="US"
          >
            United States
          </option>
          <option
            value="CA"
          >
            Canada
          </option>
          <option
            value="UK"
          >
            United Kingdom
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    class="form-section"
  >
    <h3>
      Contact Information
    </h3>
    <div
      class="form-field"
    >
      <label>
        Email
      </label>
      <input
        class="form-input"
        type="email"
        value="alice@example.com"
      />
    </div>
    <div
      class="form-field"
    >
      <label>
        Phone
      </label>
      <input
        class="form-input"
        type="tel"
        value="+1 (555) 123-4567"
      />
    </div>
  </div>
  <div
    class="form-actions"
  >
    <button
      class="btn btn-secondary btn-medium "
    >
      Cancel
    </button>
    <button
      class="btn btn-primary btn-medium "
    >
      Save Player
    </button>
  </div>
</form>
`;

exports[`Visual Regression Tests - Component Snapshots > Form Components > Tournament form - visual snapshot 1`] = `
<form
  class="tournament-form"
>
  <div
    class="form-section"
  >
    <h3>
      Basic Information
    </h3>
    <div
      class="form-field"
    >
      <label>
        Tournament Name
      </label>
      <input
        class="form-input"
        type="text"
        value="Spring Championship 2024"
      />
    </div>
    <div
      class="form-field"
    >
      <label>
        Description
      </label>
      <textarea
        class="form-textarea"
        rows="3"
      >
        Annual spring tournament
      </textarea>
    </div>
  </div>
  <div
    class="form-section"
  >
    <h3>
      Tournament Settings
    </h3>
    <div
      class="form-row"
    >
      <div
        class="form-field"
      >
        <label>
          Max Players
        </label>
        <input
          class="form-input"
          type="number"
          value="16"
        />
      </div>
      <div
        class="form-field"
      >
        <label>
          Max Rounds
        </label>
        <input
          class="form-input"
          type="number"
          value="5"
        />
      </div>
    </div>
    <div
      class="form-field"
    >
      <label>
        Pairing Method
      </label>
      <select
        class="form-select"
      >
        <option
          selected=""
          value="swiss"
        >
          Swiss System
        </option>
        <option
          value="round_robin"
        >
          Round Robin
        </option>
        <option
          value="knockout"
        >
          Knockout
        </option>
      </select>
    </div>
  </div>
  <div
    class="form-actions"
  >
    <button
      class="btn btn-secondary btn-medium "
    >
      Cancel
    </button>
    <button
      class="btn btn-primary btn-medium "
    >
      Create Tournament
    </button>
  </div>
</form>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Navigation Responsive - desktop (1024px) visual snapshot 1`] = `
<header
  class="app-header"
>
  <div
    class="nav-brand"
  >
    <h1>
      ‚ôú Pawn
    </h1>
  </div>
  <nav
    class="nav-menu"
  >
    <a
      class="nav-link active"
      href="/tournaments"
    >
      Tournaments
    </a>
    <a
      class="nav-link"
      href="/players"
    >
      Players
    </a>
    <a
      class="nav-link"
      href="/settings"
    >
      Settings
    </a>
  </nav>
</header>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Navigation Responsive - large-desktop (1440px) visual snapshot 1`] = `
<header
  class="app-header"
>
  <div
    class="nav-brand"
  >
    <h1>
      ‚ôú Pawn
    </h1>
  </div>
  <nav
    class="nav-menu"
  >
    <a
      class="nav-link active"
      href="/tournaments"
    >
      Tournaments
    </a>
    <a
      class="nav-link"
      href="/players"
    >
      Players
    </a>
    <a
      class="nav-link"
      href="/settings"
    >
      Settings
    </a>
  </nav>
</header>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Navigation Responsive - mobile (375px) visual snapshot 1`] = `
<header
  class="app-header"
>
  <div
    class="nav-brand"
  >
    <h1>
      ‚ôú Pawn
    </h1>
  </div>
  <nav
    class="nav-menu"
  >
    <a
      class="nav-link active"
      href="/tournaments"
    >
      Tournaments
    </a>
    <a
      class="nav-link"
      href="/players"
    >
      Players
    </a>
    <a
      class="nav-link"
      href="/settings"
    >
      Settings
    </a>
  </nav>
</header>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Navigation Responsive - tablet (768px) visual snapshot 1`] = `
<header
  class="app-header"
>
  <div
    class="nav-brand"
  >
    <h1>
      ‚ôú Pawn
    </h1>
  </div>
  <nav
    class="nav-menu"
  >
    <a
      class="nav-link active"
      href="/tournaments"
    >
      Tournaments
    </a>
    <a
      class="nav-link"
      href="/players"
    >
      Players
    </a>
    <a
      class="nav-link"
      href="/settings"
    >
      Settings
    </a>
  </nav>
</header>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Navigation header - visual snapshot 1`] = `
<header
  class="app-header"
>
  <div
    class="nav-brand"
  >
    <h1>
      ‚ôú Pawn
    </h1>
  </div>
  <nav
    class="nav-menu"
  >
    <a
      class="nav-link active"
      href="/tournaments"
    >
      Tournaments
    </a>
    <a
      class="nav-link"
      href="/players"
    >
      Players
    </a>
    <a
      class="nav-link"
      href="/settings"
    >
      Settings
    </a>
  </nav>
  <div
    class="nav-actions"
  >
    <button
      class="btn btn-primary btn-small "
    >
      New Tournament
    </button>
  </div>
</header>
`;

exports[`Visual Regression Tests - Component Snapshots > Layout Components > Sidebar layout - visual snapshot 1`] = `
<div
  class="app-layout"
>
  <aside
    class="sidebar"
  >
    <div
      class="sidebar-header"
    >
      <h2>
        Tournament Menu
      </h2>
    </div>
    <nav
      class="sidebar-nav"
    >
      <a
        class="sidebar-link active"
        href="#overview"
      >
        Overview
      </a>
      <a
        class="sidebar-link"
        href="#players"
      >
        Players
      </a>
      <a
        class="sidebar-link"
        href="#pairings"
      >
        Pairings
      </a>
      <a
        class="sidebar-link"
        href="#results"
      >
        Results
      </a>
      <a
        class="sidebar-link"
        href="#standings"
      >
        Standings
      </a>
    </nav>
  </aside>
  <main
    class="main-content"
  >
    <div
      class="content-header"
    >
      <h1>
        Tournament Overview
      </h1>
      <button
        class="btn btn-primary btn-medium "
      >
        Edit Tournament
      </button>
    </div>
    <div
      class="content-body"
    >
      <p>
        Main content area
      </p>
    </div>
  </main>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > LoadingButton Component > LoadingButton - idle state visual snapshot 1`] = `
<button
  class="loading-btn "
>
  <span>
    Save Tournament
  </span>
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > LoadingButton Component > LoadingButton - loading state visual snapshot 1`] = `
<button
  class="loading-btn loading"
  disabled=""
>
  <span
    class="spinner"
  />
  <span>
    Saving...
  </span>
</button>
`;

exports[`Visual Regression Tests - Component Snapshots > StatCard Component > StatCard - basic state visual snapshot 1`] = `
<div
  class="stat-card stat-card-primary"
>
  <div
    class="stat-card-header"
  >
    <h3
      class="stat-card-title"
    >
      Total Players
    </h3>
  </div>
  <div
    class="stat-card-value"
  >
    156
  </div>
  <div
    class="stat-card-subtitle"
  >
    Registered this month
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > StatCard Component > StatCard - large-numbers state visual snapshot 1`] = `
<div
  class="stat-card stat-card-primary"
>
  <div
    class="stat-card-header"
  >
    <h3
      class="stat-card-title"
    >
      Total Games Played
    </h3>
  </div>
  <div
    class="stat-card-value"
  >
    23,456
  </div>
  <div
    class="stat-card-subtitle"
  >
    Since launch
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > StatCard Component > StatCard - with-trend-negative state visual snapshot 1`] = `
<div
  class="stat-card stat-card-warning"
>
  <div
    class="stat-card-header"
  >
    <h3
      class="stat-card-title"
    >
      Completion Rate
    </h3>
    <span
      class="stat-card-trend negative"
    >
      -3
      %
    </span>
  </div>
  <div
    class="stat-card-value"
  >
    87%
  </div>
  <div
    class="stat-card-subtitle"
  >
    This week
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > StatCard Component > StatCard - with-trend-positive state visual snapshot 1`] = `
<div
  class="stat-card stat-card-success"
>
  <div
    class="stat-card-header"
  >
    <h3
      class="stat-card-title"
    >
      Active Tournaments
    </h3>
    <span
      class="stat-card-trend positive"
    >
      15
      %
    </span>
  </div>
  <div
    class="stat-card-value"
  >
    12
  </div>
  <div
    class="stat-card-subtitle"
  >
    Currently running
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > Theme Variations > Dark theme components - visual snapshot 1`] = `
<div
  class="theme-dark"
>
  <div
    class="card"
  >
    <h3>
      Dark Theme Card
    </h3>
    <p>
      This demonstrates dark theme styling
    </p>
    <div
      class="button-group"
    >
      <button
        class="btn btn-primary btn-medium "
      >
        Primary
      </button>
      <button
        class="btn btn-secondary btn-medium "
      >
        Secondary
      </button>
    </div>
  </div>
  <div
    class="stat-card stat-card-primary"
  >
    <div
      class="stat-card-header"
    >
      <h3
        class="stat-card-title"
      >
        Dark Theme Stats
      </h3>
    </div>
    <div
      class="stat-card-value"
    >
      42
    </div>
    <div
      class="stat-card-subtitle"
    >
      Looking good in dark mode
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests - Component Snapshots > Theme Variations > High contrast theme - visual snapshot 1`] = `
<div
  class="theme-high-contrast"
>
  <div
    class="card"
  >
    <h3>
      High Contrast Theme
    </h3>
    <p>
      Accessibility-focused high contrast theme
    </p>
    <div
      class="button-group"
    >
      <button
        class="btn btn-primary btn-medium "
      >
        Primary Action
      </button>
      <button
        class="btn btn-secondary btn-medium "
      >
        Secondary Action
      </button>
    </div>
  </div>
  <div
    class="empty-state"
  >
    <h3
      class="empty-state-title"
    >
      High Contrast Empty State
    </h3>
    <p
      class="empty-state-description"
    >
      Clear visibility for all users
    </p>
    <div
      class="empty-state-action"
    >
      <button
        class="btn btn-primary btn-medium "
      >
        Take Action
      </button>
    </div>
  </div>
</div>
`;
